src/MiniProfiler/ProfileProviders/WebRequestProfilerProvider.cs

```diff
@@ -150,7 +150,23 @@ private static void EnsureName(MiniProfiler profiler, HttpRequest request)
         /// </summary>
         public override MiniProfiler GetCurrentProfiler() => Current;
 
+        /// <summary>
+        /// Current head timing.
+        /// </summary>
+        public override Timing CurrentHead
```

Comment:
I'm not sure what is the intended future of WebRequestProfilerProvider and why DefaultProfilerProvider cannot be used all the time. I added this property only to support the changes to the interface. 

Another option is to "merge"  WebRequestProfilerProvider and  DefaultProfilerProvider, in this case WebRequestProfilerProvider.Current and WebRequestProfilerProvider.CurrentHead would be backed by AsyncLocals (or the polyfills). I don't see specific downside with this and the upside is that this provider would support async controller actions (that are using .ConfigureAwait(false) internally).